libafl_bolts/src/serdeany.rs

```diff
@@ -331,6 +340,27 @@ pub mod serdeany_registry {
                 .insert(type_repr_owned::<T>(), value);
         }
 
+        /// Insert a boxed element into the map if it doesn't exist, else return error.
+        #[inline]
+        pub fn try_insert_boxed<T>(&mut self, value: Box<T>) -> Result<(), Error>
+        where
+            T: crate::serdeany::SerdeAny,
+        {
+            match self.raw_entry_mut::<T>() {
```

Comment:
I just saw that the hashbrown hashmap has `try_insert`, we coudl use that directly.
The main difference is that it also returns a borrow to the existing element, which may be useful(?)